compose/project.py

```diff
@@ -245,26 +245,30 @@ def up(self,
         if force_recreate and not allow_recreate:
             raise ValueError("force_recreate and allow_recreate are in conflict")
 
-        services = self.get_services(service_names, include_deps=start_deps)
+        services = self.get_services(service_names, include_deps=False)
+        all_services = self.get_services(service_names, include_deps=start_deps)
 
-        for service in services:
+        for service in all_services:
             service.remove_duplicate_containers()
 
         plans = self._get_convergence_plans(
-            services,
+            all_services,
             allow_recreate=allow_recreate,
             force_recreate=force_recreate,
         )
 
-        return [
-            container
-            for service in services
-            for container in service.execute_convergence_plan(
+        containers_to_return = []
+
+        for service in all_services:
+            containers = service.execute_convergence_plan(
                 plans[service.name],
                 do_build=do_build,
                 timeout=timeout
             )
-        ]
+            if service in services:
+                containers_to_return += containers
+
+        return containers_to_return
```

Comment:
Ya, that's true. At some point I think it might make sense to split up some of the larger cli functions into some smaller functions (to make it easier to test), but that doesn't need to happen now.

In this case I think everything that happens after  `project.up()` could be in a separate function:

```python

if detached:
    return

print_logs_for_up(containers, service_names, monochrome, time)
```